test/Motor.Extensions.Hosting.NATS_IntegrationTest/NATSIntegrationTests.cs

```diff
@@ -17,33 +18,76 @@
         }
 
-        [Fact(Timeout = 50000)]
+        [Fact(Timeout = 50000, Skip = "does not run on ci")]
+        public async void PublishMessageWithoutException()
+        {
+            const string expectedMessage = "testMessage";
+            var topicName = _randomizerString.Generate();
+            var queueName = _randomizerString.Generate();
+
+            var publisherOptions = GetNATSBaseOptions(topicName);
+            var publisher = GetPublisher(new OptionsWrapper<NATSBaseOptions>(publisherOptions));
+
+            var consumerOptions = GetNATSConsumerOptions(topicName, queueName);
+
+            var consumer = GetConsumer<string>(new OptionsWrapper<NATSConsumerOptions>(consumerOptions), queueName);
+            var rawConsumedNatsMessage =
+                await RawConsumedNatsMessageWithNatsPublisherPublishedMessage(consumer, publisher, expectedMessage);
+
+            Assert.NotNull(rawConsumedNatsMessage);
+            Assert.Equal(expectedMessage, Encoding.UTF8.GetString(rawConsumedNatsMessage));
+        }
+
+        [Fact(Timeout = 50000, Skip = "does not run on ci")]
         public async void Consume_RawPublishIntoNATSAndConsumeCreateCloudEvent_ConsumedEqualsPublished()
         {
             const string expectedMessage = "testMessage";
             var topicName = _randomizerString.Generate();
             var queueName = _randomizerString.Generate();
-            var clientOptions = GetNATSClientOptions(topicName, queueName);
+            var clientOptions = GetNATSConsumerOptions(topicName, queueName);
 
             var nats = new NATSClientFactory().From(clientOptions);
 
-            var consumer = GetConsumer<string>(new OptionsWrapper<NATSClientOptions>(clientOptions), queueName);
+            var consumer = GetConsumer<string>(new OptionsWrapper<NATSConsumerOptions>(clientOptions), queueName);
             var rawConsumedNatsMessage = await RawConsumedNatsMessage(consumer, nats, topicName, expectedMessage);
             Assert.NotNull(rawConsumedNatsMessage);
             Assert.Equal(expectedMessage, Encoding.UTF8.GetString(rawConsumedNatsMessage));
         }
 
+        private static async Task<byte[]> RawConsumedNatsMessageWithNatsPublisherPublishedMessage(
+            NATSConsumer<string> consumer, NATSPublisher publisher, string expectedMessage)
+        {
+            var rawConsumedNatsMessage = (byte[])null;
+            var taskCompletionSource = new TaskCompletionSource();
+            consumer.ConsumeCallbackAsync = async (dataEvent, _) =>
+            {
+                rawConsumedNatsMessage = dataEvent.TypedData;
+                taskCompletionSource.TrySetResult();
+                return await Task.FromResult(ProcessedMessageStatus.Success);
+            };
+
+            await consumer.StartAsync();
+            var consumerStartTask = consumer.ExecuteAsync();
+
+            await Task.Delay(TimeSpan.FromSeconds(5));
+            await publisher.PublishMessageAsync(
```

Comment:
(nitpick) If I'm not mistaken, then this is the only line that differs in this method from `RawConsumedNatsMessage`. Can't the two methods be combined into one? I think that would make it also easier to understand that both tests are basically the same with the only difference that one uses the `NATSPublisher` whereas the other one implements the publishing itself to test the consuming of messages.

Reply:
i thought that was clear by the given function-names